File: functions/functions2/ex11.c
/*Write a function to check whether a number is perfect or not.*/
#include <stdio.h>

/* say if a  number is perfect */
void is_perfect(int num);

int main()
{
    int number = 0;
    printf("Introduce a number (int): ");
    scanf("%i",&number);

    is_perfect(number);

    return 0;
}

void is_perfect(int num)
{
    int sum = 0;

    for (int i = 1; i < num; i++)
    {
        if ( (num%i) == 0)
        {
            sum = sum + i;
        }
    }
    
    if (sum == num)
    {
        printf("%i is perfect\n",num);
    }
    else
    {
        printf("%i is not perfect\n",num);
    }
}

File: functions/functions2/ex11.py
"""Write a function to check whetear a number is perfect or not"""

def is_perfect(number):
    aux = 0

    for i in range(1,number):
        if (number%i) == 0:
            aux += i
    return "is perfect" if aux == number else "is not perfect"

#-------------------------------------------------

num = int(input("Insert a number: "))

print(num,is_perfect(num))
